lib/cosmos/topics/limits_event_topic.rb

```diff
@@ -21,15 +21,31 @@
 
 module Cosmos
   class LimitsEventTopic < Topic
-    def self.write(type:, target_name:, packet_name:, item_name:, old_limits_state:, new_limits_state:, time_nsec:, message:, scope:)
-      Store.instance.write_topic("#{scope}__cosmos_limits_events",
-        {type: 'LIMITS_CHANGE', target_name: target_name, packet_name: packet_name,
+    def self.write(target_name, packet_name, item_name, old_limits_state, new_limits_state, time_nsec, message, type:, scope:)
+      Store.write_topic("#{scope}__cosmos_limits_events",
+        {type: type, target_name: target_name, packet_name: packet_name,
           item_name: item_name, old_limits_state: old_limits_state, new_limits_state: new_limits_state,
           time_nsec: time_nsec, message: message})
       # The current_limits hash keeps only the current limits state of items
       # It is used by the API to determine the overall limits state
       # TODO: How do we maintain / clean this hash?
       Store.hset("#{scope}__current_limits","#{target_name}__#{packet_name}__#{item_name}", new_limits_state)
     end
+
+    def self.read(offset = nil, count: 100, scope:)
+      topic = "#{scope}__cosmos_limits_events"
+      if offset
+        result = Store.xread(topic, offset, count: count)
+        if result.empty?
+          [] # We want to return an empty array rather than an empty hash
+        else
+          # result is a hash with the topic key followed by an array of results
+          # This returns just the array of arrays [[offset, hash], [offset, hash], ...]
+          result[topic]
+        end
+      else
+        Store.xrevrange(topic, count: 1)
+      end
+    end
```

Comment:
If you don't pass in an offset I use xrevrange to grab the last limits event (which also returns the offset). At that point you can pass in your offset and it uses regular xread to return up to 100 limits events (by default).